src/protocol/jms/org/apache/jmeter/protocol/jms/sampler/JMSSampler.java

```diff
@@ -142,48 +173,244 @@ public SampleResult sample(Entry entry) {
+                    MapMessage mapm = (MapMessage) msg;
+                    @SuppressWarnings("unchecked") // MapNames are Strings
+                    Enumeration<String> enumb = mapm.getMapNames();
+                    while (enumb.hasMoreElements()) {
+                        String name = enumb.nextElement();
+                        Object obj = mapm.getObject(name);
+                        buffer.append(name);
+                        buffer.append(",");
+                        buffer.append(obj.getClass().getCanonicalName());
+                        buffer.append(",");
+                        buffer.append(obj);
+                        buffer.append("\n");
+                    }
+                }
+                Utils.messageProperties(propBuffer, msg);
+            } catch (JMSException e) {
+                LOGGER.error(e.getMessage());
+            }
+        }
+    }
+
+    private String browseQueueDetails(Queue queue, SampleResult res) {
+        try {
+            String messageBodies = new String("\n==== Browsing Messages === \n");
+            // get some queue details
+            QueueBrowser qBrowser = session.createBrowser(queue);
+            // browse the messages
+            Enumeration<?> e = qBrowser.getEnumeration();
+            int numMsgs = 0;
+            // count number of messages
+            String corrID = "";
+            while (e.hasMoreElements()) {
+                TextMessage message = (TextMessage) e.nextElement();
+                corrID = message.getJMSCorrelationID();
+                if (corrID == null) {
+                    corrID = message.getJMSMessageID();
+                    messageBodies = messageBodies + numMsgs + " - MessageID: " + corrID + ": " + message.getText()
+                            + "\n";
+                } else {
+                    messageBodies = messageBodies + numMsgs + " - CorrelationID: " + corrID + ": " + message.getText()
+                            + "\n";
+                }
+                numMsgs++;
+            }
+            res.setResponseMessage(numMsgs + " messages available on the queue");
+            res.setResponseHeaders(qBrowser.toString());
+            return (messageBodies + queue.getQueueName() + " has " + numMsgs + " messages");
+        } catch (Exception e) {
+            res.setResponseMessage("Error counting message on the queue");
+            e.printStackTrace();
+            LOGGER.error(e.getMessage());
+            return "";
+        }
+    }
+
+    private String clearQueue(Queue queue, SampleResult res) {
+        String retVal = null;
+        try {
+            QueueReceiver consumer = session.createReceiver(queue);
+            Message deletedMsg = null;
+            long deletedMsgCount = 0;
+            do {
+                deletedMsg = consumer.receiveNoWait();
+                if (deletedMsg != null) {
+                    deletedMsgCount++;
+                    deletedMsg.acknowledge();
```

Comment:
Shouldn't this be parameterized ? There are 4 modes, if we ACK for AUTO_ACK that would be wrong no ?